databroker/broker.py

```diff
@@ -470,6 +490,10 @@ def fill_event(self, event, handler_registry=None, handler_overrides=None):
         """
         Populate events with externally stored data.
 
+        .. warning
+
+            This mutates the event's ``data`` field in-place
```

Comment:
This warning does not match with what happens below. Did you mean to pass `inplace=True` below?